numpy/core/_add_newdocs.py

```diff
@@ -914,6 +925,16 @@
         Whether to store multi-dimensional data in row-major
         (C-style) or column-major (Fortran-style) order in
         memory.
+    like : array_like
+        Reference array-like object created by a downstream library, such
+        as CuPy, Dask, xarray, etc. If specified, the function will be
```

Comment:
Ideally, something like this would suffice:

> Reference array to copy the type and possibly metadata from.

But I'd lobe to hear more thoughts about possible wording (or lack thereof).

Reply:
I'm more worried if users will understand what "downstream library" or "metadata" means in this context, that's why I tried to add examples. I wasn't able to find terminology that would be less loaded of technical jargon. With that said, I think we may want to rephrase that in a way that would be easy to understand for someone with little knowledge of those terms. Any suggestions are appreciated. :)

Reply:
"Array libraries other than NumPy itself"

Reply:
I think it would be good to reference/link to NEP 35 for more details. Maybe something like:

> Reference array-like object that may override the implementation of this function if it implements an `__array_function__` method. See NEP 35 for details: https://numpy.org/neps/nep-0035-array-creation-dispatch-with-array-function.html

Reply:
I actually like having concrete examples listed as Peter did here 👍 Just my two cents...

Reply:
In the meantime, I also suggested a different writing in https://github.com/numpy/numpy/pull/16935#discussion_r465311220:

```
Reference object created by a library implementing ndarray-like API.
If specified, the function will be dispatched to the library implementing
the reference object, provided it supports the __array_function__ protocol.
```

That said, I have no strong opinions either way, I like @shoyer 's  suggestion as well. It seems though that there are two people who like also having explicit examples. I'm fine either way, but I'm going to way for a bit before doing any changes, perhaps more people will chime in.

Reply:
@seberg I'm pinging you and linking your https://github.com/numpy/numpy/pull/16935#discussion_r465859171 here. This thread is related to your comment but you may have missed it and would be interesting to know what you think of the suggestions here as well.

Reply:
Yeah, the quote you had above sounds good. Tried to brainstorm whether we can dumb it down even more, but take it or leave it. (github orders comments a bit oddly to show them both as part of the review and as a reply to the conversation when replying in a "review".)

Reply:
Ok, so it seems we have a few alternatives:

1. Keep it as is;
1. Use @shoyer 's suggestion in https://github.com/numpy/numpy/pull/16935/files#r465308481 ;
1. Use my suggestion in https://github.com/numpy/numpy/pull/16935/files#r465320170 ;
1. Something else?

I don't know how to fairly choose so let's do it as kids would and please vote (only once) by adding an emoji to this comment: 👍  for option 1, 🎉  for 2, ❤️  for 3 and 🚀  for 4 (this requires also a suggestion!).

Reply:
Voting for your suggestion. I am pretty happy with most wordings, they may need fixing up, buts its OK to do that later. adding the brainstorming suggestion:

> Reference object to allow the creation of arrays which are not NumPy arrays. If an array-like passed in as `like` supports the ``__array_function__`` protocol, the result will be defined by it. In this case, it ensures the creation of an array object compatible with that passed in as `like`.

Reply:
I updated the docstring in https://github.com/numpy/numpy/pull/16935/commits/eed6ec585d76a7d09b54b191d6674ad8e3f37c72 with @seberg 's suggestion and a minor modification at the end to avoid being repetitive with the `like`. Please let us know if you disagree with the current reading or if you have suggestions to improve it.

Reply:
Resolving this thread, I made a new comment on the current wording: https://github.com/numpy/numpy/pull/16935#discussion_r468164478